src/app/compiler.js

```diff
@@ -189,7 +200,9 @@ function Compiler (editor, queryParams, handleGithubCall, updateFiles) {
       compilationFinished({ error: 'Worker error: ' + msg.data });
     });
     compileJSON = function (source, optimize) {
-      worker.postMessage({cmd: 'compile', source: JSON.stringify(source), optimize: optimize});
+      jobs[nextJobId] = {sources: source};
```

Comment:
Oh you are using delete. Well actually that works and you'll have empty slots in JS (just like in Lua).